classes/mail/EmailTemplateDAO.inc.php

```diff
@@ -213,6 +206,60 @@ function installEmailTemplates($templatesFile, $returnSql = false, $emailKey = n
 			if (!$returnSql) {
 				$this->update(array_shift($sql));
 			}
+
+			// Add localized data
+			$additionalQueries = $this->installEmailTemplateLocaleData($templatesFile, $locales, $returnSql, $attrs['key']);
+			if ($returnSql) $sql = array_merge($sql, $additionalQueries);
+		}
+		if ($returnSql) return $sql;
+		return true;
+	}
+
+	/**
+	 * Install email template contents from an XML file.
+	 * NOTE: Uses qstr instead of ? bindings so that SQL can be fetched
+	 * rather than executed.
+	 * @param $templatesFile string Filename to install
+	 * @param $locales List of locales to install data for
+	 * @param $returnSql boolean Whether or not to return SQL rather than
+	 * executing it
+	 * @param $emailKey string Optional name of single email key to install,
+	 * skipping others
+	 * @return array|boolean
+	 */
+	function installEmailTemplateLocaleData($templatesFile, $locales = array(), $returnSql = false, $emailKey = null) {
+		$xmlDao = new XMLDAO();
+		$sql = array();
+		$data = $xmlDao->parseStruct($templatesFile, array('email'));
+		if (!isset($data['email'])) return false;
+		$dataSource = $this->getDataSource();
+		foreach ($data['email'] as $entry) {
+			$attrs = $entry['attributes'];
+			if ($emailKey && $emailKey != $attrs['key']) continue;
+
+			$subject = $attrs['subject']??null;
+			$body = $attrs['body']??null;
```

Comment:
Just an idle comment on code style, but how do you feel about the distinction between `$a??$b` and `$a ?? $b`? The space helps me see the syntax a little bit quicker, especially in cases where there's more than just `$a` and `$b`, like here with the array key and `null`.

Reply:
No major preference here -- I'm happy to go woth `$a ?? $b`.